experimenter/experimenter/experiments/api/v5/serializers.py

```diff
@@ -1531,6 +1536,79 @@ def _validate_bucket_duplicates(self, data):
 
         return data
 
+    def _validate_localization_content(self, data):
+        is_localized = data.get("is_localized")
+
+        if not is_localized:
+            return data
+
+        localized_content = data.get("localized_content")
+        locales = data.get("locales")
+        min_version = data.get("firefox_min_version", "")
+        if min_version == "" or (
+            NimbusExperiment.Version.parse(min_version)
+            < NimbusExperiment.Version.parse(NimbusExperiment.Version.FIREFOX_113)
+        ):
+            raise serializers.ValidationError(
+                {
+                    "firefox_min_version": [
+                        "Firefox version must not be < 113 for localized experiments."
+                    ],
+                }
+            )
+        if not locales:
+            raise serializers.ValidationError(
+                {"locales": "Locales must not be empty for a localized experiment."}
+            )
+
+        locales = [locale.code for locale in locales]
+
+        try:
+            json_content = json.loads(localized_content)
+        except Exception as e:
+            raise serializers.ValidationError({"localized_content": f"Invalid JSON: {e}"})
+
+        path = (
+            files("mozilla_nimbus_shared")
+            / "schemas"
+            / "experiments"
+            / "NimbusExperiment.json"
+        )
+        schema = json.loads(path.read_text())
+        schema = schema["definitions"]["NimbusExperiment"]["properties"]["localizations"]
+
+        if is_localized and localized_content:
+            try:
+                jsonschema.validate(json_content, schema)
+            except Exception as e:
+                raise serializers.ValidationError(
+                    {"localized_content": f"Localization schema error: {e}"}
+                )
+
+            for locale in locales:
+                if locale not in json_content:
+                    raise serializers.ValidationError(
+                        {
+                            "localized_content": (
+                                f"Experiment locale {locale} not present "
+                                f"in localizations."
+                            )
+                        }
+                    )
+
+            for localization in json_content:
+                if localization not in locales:
+                    raise serializers.ValidationError(
+                        {
+                            "localized_content": (
+                                f"Localization locale {localization} "
+                                f"does not exist in experiment locales."
+                            )
+                        }
+                    )
+
```

Comment:
There is a piece missing here from the proposal:

> Experimenter must perform schema validation with every localization entry present and all of them must succeed

That means we need to do the following algorithm:

```
for locale in locales:
  for branch in branches:
    for feature in branch.features:
      value <- substitute feature.value with localizations
      validate value against feature schema
```

The algorithm used in desktop to do substitution can be found in [`_ExperimentFeature#substituteLocalizations`](https://searchfox.org/mozilla-central/rev/31f5847a4494b3646edabbdd7ea39cb88509afe2/toolkit/components/nimbus/ExperimentAPI.sys.mjs#522)

This will need a test where one localizations entry is missing an ID that is required to perform the substitution.

e.g.

```
value = { "foo": { "$l10n": { "id": "foo", "text": "foo text", "comment": "foo comment" } } }
localizations = {
  "en-US": { "foo": "translated foo" },
  "en-CA": {}
}
```

which should throw an error for the en-CA case.